src/grt/grt-wave_options-parse.adb

```diff
@@ -0,0 +1,318 @@
+      end if;
+
+      --~ Debug.Dump_Tree;
+
+   end Start;
+
+-- private --------------------------------------------------------------------
+
+   procedure Parse_Version (Line : String_Acc) is
+      Msg_Invalid_Format : constant String := "invalid version format";
+      First, Dot_Index, Num : Integer;
+   begin
+
+      if Version /= (others => -1) then
+         Error_Context ("version is set more than once");
+      end if;
+
+      if Trees /= Tree_Array'(others => null) then
+         Error_Context ("version cannot be set after signal paths");
+      end if;
+
+      First := First_Non_Whitespace_Pos (Line (Line'First + 1 .. Line'Last));
+      if Line (First .. First + 6) /= "version" then
+         Error_Context (Msg_Invalid_Format);
+      end if;
+
+      First := First + 7;
+      if not Is_Whitespace (Line (First)) then --catch "version\n","version1.0"
+         Error_Context (Msg_Invalid_Format);
+      end if;
+
+      First := First_Non_Whitespace_Pos (Line (First + 1 .. Line'Last));
+      if First = -1 then -- catch "version \n", "version  \n", etc
+         Error_Context (Msg_Invalid_Format);
+      end if;
+
+      Dot_Index := Find (Line (First + 1 .. Line'Last), '.');
+      if Dot_Index = -1 then -- catch the absence of "." or "version ."
+         Error_Context (Msg_Invalid_Format);
+      end if;
+
+      Num := Value (Line (First .. Dot_Index - 1));
+      if Num = -1 then -- catch version a.2
+         Error_Context (Msg_Invalid_Format);
+      end if;
+      Version.Major := Num;
+
+      Num := Value (Line (Dot_Index + 1 .. Line'Last));
+      if Num = -1 then -- catch version 1.a
+         Error_Context (Msg_Invalid_Format);
+      end if;
+      Version.Minor := Num;
+
+      if Version.Major /= Current_Version.Major
+        or else Version.Minor > Current_Version.Minor
+      then
+         Print_Context (Error);
+         Error_C ("unsupported format version; it must be ");
+         if Current_Version.Minor /= 0 then
+            Error_C ("between ");
+            Print_Version (Version_Type'(Current_Version.Major, 0));
+            Error_C (" and ");
+         end if;
+         Print_Version (Current_Version);
+         Error_E;
+      end if;
+
+   end Parse_Version;
+
+   procedure Print_Version (Version : Version_Type) is
+      Num_Str : String (1 .. Value_String_Size);
+      First : Positive;
+   begin
+      To_String (Num_Str, First, Ghdl_I32 (Version.Major));
+      Report_C (Num_Str (First .. Num_Str'Last));
+      Report_C (".");
+      To_String (Num_Str, First, Ghdl_I32 (Version.Minor));
+      Report_C (Num_Str (First .. Num_Str'Last));
+   end Print_Version;
+
+   --------------------------------------------------------------------------
+
+   procedure Parse_Path (Line : String_Acc) is
+      First, Last : Positive;
+      Tree_Updated : Boolean;
+      Tree_Index : Tree_Index_Type;
+   begin
+      To_Lower (Line_Context.Str.all);
```

Comment:
There is one problem if I don't do the conversion to lower here :
Suppose we have the following :

/top/a
/TOP/b

then without a conversion to lower, the tree will looks like :
Tree ptr --> top --> a
          |-> TOP --> b

instead of :
Tree ptr --> top --> a
                  |-> b

If in the future we add support to verilog and system-C. I see 2 solutions :

1) - Create 2 trees when parsing : one case sensitive and one case 
insensitive
    - Then latter when we have more informations, prune VHDL paths from 
the case
      sensitive tree and prune verilog / system-C paths from the case
      insensitive tree (maybe it's not really needed)
    - Use the right tree while looking for signals to be displayed in 
the design

2) - Create only 1 case sensitive tree
    - Then latter, look for VHDL paths in the tree and merge elements 
who have
      the same name after lowering their characters.
    - No change should be afterwards needed in the code that find 
signals to be
      displayed

I have the impression that the first approach is slighty easier than the
second, and currently the code match well the first approach. Anyways, I 
would
prefer not add complexity before it's time and maybe in the futur I may 
need to
do code rewrites before we had support for verilog and system-C and then 
need
to have a different approach.

Nevertheless, we still have in mind that the case sensitivity is an 
issue that
needs to be fixed some time in the future :).